### DISCUSSION TITLE
Retrieving weather data takes 8 minutes

### DISCUSSION THREAD

**Discussion by michaeloconnor1717** [weather-gov/api#310] [Q&A]
I am trying to use this API to retrieve weather forecast via an HTTPS request from an ESP8266 using the Arduino IDE.

While it gets the JSON pretty quickly it things seems to keep receiving data from the request for 8 more minutes but does not display or show any data being returned.
I know that the esp8266 can handle these requests as I have done similar things, so i am hoping i am just not understanding the proper way to make a forecast request.

One thing to note is that no where in the program is it supposed to send a blank line to the terminal every 5 seconds, which is essentially whats happening for 8 minutes, and thats why i wanted to see if I am incorrectly calling the request.

Let me know if theres more information I can provide to clarify anything.

TLDR version
here's the simplified version of the request I'm using:
api.weather.gov/gridpoints/LZK/49,94/forecast
and its taking a really long time

Esp8266  Code
```
#include <Arduino.h>
#include <ESP8266WiFi.h>
#include "HTTPSRedirect.h"
#include <SoftwareSerial.h>
#include <NTPClient.h>
#include <WiFiUdp.h>

HTTPSRedirect* weather = nullptr;
const char* host1 = "api.weather.gov";
String url1 = "/gridpoints/LZK/49,94/forecast";

SoftwareSerial espToMega(16, 5);

// Define NTP Client to get time
#define utcOffsetInSeconds -18000
WiFiUDP ntpUDP;

NTPClient timeClient(ntpUDP, "pool.ntp.org", utcOffsetInSeconds);
void setup() {
  pinMode(10, INPUT);
  espToMega.begin(115200);
  Serial.begin(9600);
  delay(10);
  Serial.println('\n');

  // Connect to WiFi
  WiFi.begin(ssid, password);
  Serial.print("Connecting to ");
  Serial.print(ssid); Serial.println(" ...");

  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.print(".");
  }

  Serial.println('\n');
  Serial.println("Connection established!");
  Serial.print("IP address:\t");
  Serial.println(WiFi.localIP());

  weather = new HTTPSRedirect(httpsPort);
  weather->setInsecure();
  weather->setPrintResponseBody(true);
  weather->setContentTypeHeader("application/json");

  espToMega.println("Connecting");
  Serial.print("Connecting to ");
  Serial.println(host1);

  bool flag1 = false;
  for (int i = 0; i < 5; i++) {
    int retval1 = weather->connect(host1, httpsPort);
    if (retval1 == 1) {
      flag1 = true;
      Serial.println("Connected");
      break;
    }
    else
      Serial.println("Connection failed. Retrying...");
  }

  if (!flag1) {
    Serial.print("Could not connect to server: ");
    Serial.println(host1);
    return;
  }
  String events = "";
  espToMega.println("clear");
  espToMega.flush();
  delay(500);
  // delete HTTPSRedirect object
  delete weather;
  weather = nullptr;

  timeClient.begin();
  timeClient.update();



  static bool flag = false;
  if (!flag) {
    weather = new HTTPSRedirect(httpsPort);
    weather->setInsecure();
    flag = true;
    weather->setPrintResponseBody(true);
    weather->setContentTypeHeader("application/json");
  }
  if (weather != nullptr) {
    if (!weather->connected()) {
      weather->connect(host1, httpsPort);
    }
  }
  else {
    Serial.println("Error creating client object!");
  }

  //getting data
  Serial.println("requesting data...");

  if (weather->GET(url1, host1)) {
    // do stuff here if publish was successful
  }
  else {
    // do stuff here if publish was not successful
    Serial.println("Error while connecting");
  }

  events = weather->getResponseBody();
  Serial.print((String)("Events: " + events));
  Serial.flush();
  espToMega.print(events);
  Serial.println("Done with Setup");
}
```

Serial Output
```
19:40:45.675 -> Connection established!
19:40:45.710 -> IP address:	192.168.1.15
19:40:45.744 -> Connecting to api.weather.gov
19:40:46.800 -> Connected
19:40:48.433 -> requesting data...
19:40:48.501 -> {
19:40:48.501 ->     "@context": [
19:40:48.535 ->         "https://geojson.org/geojson-ld/geojson-context.jsonld",
19:40:48.603 ->         {
19:40:48.603 ->             "@version": "1.1",
19:40:48.637 ->             "wx": "https://api.weather.gov/ontology#",
19:40:48.705 ->             "geo": "http://www.opengis.net/ont/geosparql#",
19:40:48.773 ->             "unit": "http://codes.wmo.int/common/unit/",
19:40:48.841 ->             "@vocab": "https://api.weather.gov/ontology#"
19:40:48.875 ->         }
19:40:48.909 ->     ],
19:40:48.909 ->     "type": "Feature",
19:40:48.943 ->     "geometry": {
19:40:48.943 ->         "type": "Polygon",
19:40:48.977 ->         "coordinates": [
19:40:49.011 ->             [
19:40:49.011 ->                 [
19:40:49.045 ->                     -93.170613200000005,
19:40:49.079 ->                     35.271125400000003
19:40:49.113 ->                 ],
19:40:49.147 ->                 [
19:40:49.181 ->                     -93.170984500000003,
19:40:49.215 ->                     35.248660200000003
19:40:49.249 ->                 ],
19:40:49.283 ->                 [
19:40:49.283 ->                     -93.1434876,
19:40:49.317 ->                     35.248355000000004
19:40:49.385 ->                 ],
19:40:49.385 ->                 [
19:40:49.420 ->                     -93.143110800000002,
19:40:49.454 ->                     35.270820100000002
19:40:49.488 ->                 ],
19:40:49.522 ->                 [
19:40:49.522 ->                     -93.170613200000005,
19:40:49.590 ->                     35.271125400000003
19:40:49.624 ->                 ]
19:40:49.658 ->             ]
19:40:49.658 ->         ]
19:40:49.658 ->     },
19:40:49.692 ->     "properties": {
19:40:49.692 ->         "updated": "2020-07-25T23:36:17+00:00",
19:40:49.760 ->         "units": "us",
19:40:49.794 ->         "forecastGenerator": "BaselineForecastGenerator",
19:40:49.828 ->         "generatedAt": "2020-07-26T00:31:24+00:00",
19:40:49.896 ->         "updateTime": "2020-07-25T23:36:17+00:00",
19:40:49.964 ->         "validTimes": "2020-07-25T17:00:00+00:00/P7DT20H",
19:40:49.998 ->         "elevation": {
19:40:50.032 ->             "value": 146.9136,
19:40:50.066 ->             "unitCode": "unit:m"
19:40:50.100 ->         },
19:40:50.100 ->         "periods": [
19:40:50.134 ->             {
19:40:50.168 ->                 "number": 1,
19:40:50.202 ->                 "name": "Tonight",
19:40:50.236 ->                 "startTime": "2020-07-25T19:00:00-05:00",
19:40:50.270 ->                 "endTime": "2020-07-26T06:00:00-05:00",
19:40:50.338 ->                 "isDaytime": false,
19:40:50.372 ->                 "temperature": 74,
19:40:50.406 ->                 "temperatureUnit": "F",
19:40:50.474 ->                 "temperatureTrend": null,
19:40:50.508 ->                 "windSpeed": "0 to 5 mph",
19:40:50.542 ->                 "windDirection": "ESE",
19:40:50.610 ->                 "icon": "https://api.weather.gov/icons/land/night/sct,20/sct?size=medium",
19:40:50.678 ->                 "shortForecast": "Partly Cloudy",
19:40:50.746 ->                 "detailedForecast": "Partly cloudy, with a low around 74. Heat index values as high as 98. East southeast wind 0 to 5 mph. Chance of precipitation is 20%. New rainfall amounts less than a tenth of an inch possible."
19:40:50.984 ->             },
19:40:51.018 ->             {
19:40:51.018 ->                 "number": 2,
19:40:51.052 ->                 "name": "Sunday",
19:40:51.086 ->                 "startTime": "2020-07-26T06:00:00-05:00",
19:40:51.154 ->                 "endTime": "2020-07-26T18:00:00-05:00",
19:40:51.222 ->                 "isDaytime": true,
19:40:51.256 ->                 "temperature": 92,
19:40:51.290 ->                 "temperatureUnit": "F",
19:40:51.324 ->                 "temperatureTrend": null,
19:40:51.358 ->                 "windSpeed": "0 to 5 mph",
19:40:51.426 ->                 "windDirection": "SE",
19:40:51.460 ->                 "icon": "https://api.weather.gov/icons/land/day/sct/sct,20?size=medium",
19:40:51.562 ->                 "shortForecast": "Mostly Sunny",
19:40:51.596 ->                 "detailedForecast": "Mostly sunny, with a high near 92. Heat index values as high as 99. Southeast wind 0 to 5 mph. Chance of precipitation is 20%. New rainfall amounts less than a tenth of an inch possible."
19:40:51.834 ->             },
19:40:51.868 ->             {
19:40:51.868 ->                 "number": 3,
19:40:51.902 ->                 "name": "Sunday Night",
19:40:51.936 ->                 "startTime": "2020-07-26T18:00:00-05:00",
19:40:52.005 ->                 "endTime": "2020-07-27T06:00:00-05:00",
19:40:52.073 ->                 "isDaytime": false,
19:40:52.107 ->                 "temperature": 73,
19:40:52.141 ->                 "temperatureUnit": "F",
19:40:52.175 ->                 "temperatureTrend": null,
19:40:52.243 ->                 "windSpeed": "0 to 5 mph",
19:40:52.277 ->                 "windDirection": "S",
19:40:52.311 ->                 "icon": "https://api.weather.gov/icons/land/night/few,20/few?size=medium",
19:40:52.415 ->                 "shortForecast": "Mostly Clear",
19:40:52.484 ->                 "detailedForecast": "Mostly clear, with a low around 73. Heat index values as high as 98. South wind 0 to 5 mph. Chance of precipitation is 20%. New rainfall amounts less than a tenth of an inch possible."
19:40:52.688 ->             },
19:40:52.722 ->             {
19:40:52.722 ->                 "number": 4,
19:40:52.756 ->                 "name": "Monday",
19:40:52.790 ->                 "startTime": "2020-07-27T06:00:00-05:00",
19:40:52.858 ->                 "endTime": "2020-07-27T18:00:00-05:00",
19:40:52.926 ->                 "isDaytime": true,
19:40:52.960 ->                 "temperature": 93,
19:40:52.994 ->                 "temperatureUnit": "F",
19:40:53.028 ->                 "temperatureTrend": null,
19:40:53.096 ->                 "windSpeed": "0 to 5 mph",
19:40:53.130 ->                 "windDirection": "SSW",
19:40:53.164 ->                 "icon": "https://api.weather.gov/icons/land/day/sct/tsra_hi,30?size=medium",
19:40:53.266 ->                 "shortForecast": "Mostly Sunny then Chance Showers And Thunderstorms",
19:40:53.368 ->                 "detailedForecast": "A chance of showers and thunderstorms after 4pm. Mostly sunny, with a high near 93. Heat index values as high as 100. South southwest wind 0 to 5 mph. Chance of precipitation is 30%."
19:40:53.606 ->             },
19:40:53.606 ->             {
19:40:53.606 ->                 "number": 5,
19:40:53.640 ->                 "name": "Monday Night",
19:40:53.708 ->                 "startTime": "2020-07-27T18:00:00-05:00",
19:40:53.742 ->                 "endTime": "2020-07-28T06:00:00-05:00",
19:40:53.810 ->                 "isDaytime": false,
19:40:53.844 ->                 "temperature": 72,
19:40:53.878 ->                 "temperatureUnit": "F",
19:40:53.946 ->                 "temperatureTrend": null,
19:40:53.980 ->                 "windSpeed": "0 to 5 mph",
19:40:54.014 ->                 "windDirection": "SSW",
19:40:54.082 ->                 "icon": "https://api.weather.gov/icons/land/night/tsra_hi,30/tsra_hi,40?size=medium",
19:40:54.184 ->                 "shortForecast": "Chance Showers And Thunderstorms",
19:40:54.252 ->                 "detailedForecast": "A chance of showers and thunderstorms. Mostly cloudy, with a low around 72. South southwest wind 0 to 5 mph. Chance of precipitation is 40%."
19:40:54.423 ->             },
19:40:54.457 ->             {
19:40:54.457 ->                 "number": 6,
19:40:54.491 ->                 "name": "Tuesday",
19:40:54.525 ->                 "startTime": "2020-07-28T06:00:00-05:00",
19:40:54.593 ->                 "endTime": "2020-07-28T18:00:00-05:00",
19:40:54.661 ->                 "isDaytime": true,
19:40:54.695 ->                 "temperature": 89,
19:40:54.729 ->                 "temperatureUnit": "F",
19:40:54.763 ->                 "temperatureTrend": null,
19:40:54.831 ->                 "windSpeed": "0 to 5 mph",
19:40:54.865 ->                 "windDirection": "SSE",
19:40:54.899 ->                 "icon": "https://api.weather.gov/icons/land/day/tsra_sct,60?size=medium",
19:40:55.001 ->                 "shortForecast": "Showers And Thunderstorms Likely",
19:40:55.069 ->                 "detailedForecast": "Showers and thunderstorms likely. Partly sunny, with a high near 89. South southeast wind 0 to 5 mph. Chance of precipitation is 60%."
19:40:55.274 ->             },
19:40:55.274 ->             {
19:40:55.274 ->                 "number": 7,
19:40:55.308 ->                 "name": "Tuesday Night",
19:40:55.378 ->                 "startTime": "2020-07-28T18:00:00-05:00",
19:40:55.412 ->                 "endTime": "2020-07-29T06:00:00-05:00",
19:40:55.480 ->                 "isDaytime": false,
19:40:55.514 ->                 "temperature": 72,
19:40:55.548 ->                 "temperatureUnit": "F",
19:40:55.615 ->                 "temperatureTrend": null,
19:40:55.649 ->                 "windSpeed": "0 to 5 mph",
19:40:55.683 ->                 "windDirection": "ESE",
19:40:55.751 ->                 "icon": "https://api.weather.gov/icons/land/night/tsra_sct,60/tsra_sct,30?size=medium",
19:40:55.854 ->                 "shortForecast": "Showers And Thunderstorms Likely",
19:40:55.922 ->                 "detailedForecast": "Showers and thunderstorms likely. Mostly cloudy, with a low around 72. East southeast wind 0 to 5 mph. Chance of precipitation is 60%."
19:40:56.094 ->             },
19:40:56.128 ->             {
19:40:56.128 ->                 "number": 8,
19:40:56.162 ->                 "name": "Wednesday",
19:40:56.196 ->                 "startTime": "2020-07-29T06:00:00-05:00",
19:40:56.264 ->                 "endTime": "2020-07-29T18:00:00-05:00",
19:40:56.332 ->                 "isDaytime": true,
19:40:56.366 ->                 "temperature": 85,
19:40:56.400 ->                 "temperatureUnit": "F",
19:40:56.434 ->                 "temperatureTrend": null,
19:40:56.502 ->                 "windSpeed": "0 to 5 mph",
19:40:56.536 ->                 "windDirection": "SSE",
19:40:56.570 ->                 "icon": "https://api.weather.gov/icons/land/day/tsra,60?size=medium",
19:40:56.672 ->                 "shortForecast": "Showers And Thunderstorms Likely",
19:40:56.740 ->                 "detailedForecast": "Showers and thunderstorms likely. Mostly cloudy, with a high near 85. South southeast wind 0 to 5 mph. Chance of precipitation is 60%."
19:40:56.910 ->             },
19:40:56.944 ->             {
19:40:56.944 ->                 "number": 9,
19:40:56.978 ->                 "name": "Wednesday Night",
19:40:57.046 ->                 "startTime": "2020-07-29T18:00:00-05:00",
19:40:57.080 ->                 "endTime": "2020-07-30T06:00:00-05:00",
19:40:57.148 ->                 "isDaytime": false,
19:40:57.183 ->                 "temperature": 71,
19:40:57.217 ->                 "temperatureUnit": "F",
19:40:57.286 ->                 "temperatureTrend": null,
19:40:57.320 ->                 "windSpeed": "5 mph",
19:40:57.354 ->                 "windDirection": "SSW",
19:40:57.388 ->                 "icon": "https://api.weather.gov/icons/land/night/tsra,60/tsra,40?size=medium",
19:40:57.490 ->                 "shortForecast": "Showers And Thunderstorms Likely",
19:40:57.558 ->                 "detailedForecast": "Showers and thunderstorms likely. Mostly cloudy, with a low around 71. South southwest wind around 5 mph. Chance of precipitation is 60%."
19:40:57.762 ->             },
19:40:57.762 ->             {
19:40:57.796 ->                 "number": 10,
19:40:57.830 ->                 "name": "Thursday",
19:40:57.864 ->                 "startTime": "2020-07-30T06:00:00-05:00",
19:40:57.932 ->                 "endTime": "2020-07-30T18:00:00-05:00",
19:40:57.966 ->                 "isDaytime": true,
19:40:58.034 ->                 "temperature": 88,
19:40:58.068 ->                 "temperatureUnit": "F",
19:40:58.102 ->                 "temperatureTrend": null,
19:40:58.136 ->                 "windSpeed": "5 mph",
19:40:58.170 ->                 "windDirection": "SW",
19:40:58.238 ->                 "icon": "https://api.weather.gov/icons/land/day/tsra,60?size=medium",
19:40:58.306 ->                 "shortForecast": "Showers And Thunderstorms Likely",
19:40:58.374 ->                 "detailedForecast": "Showers and thunderstorms likely. Mostly cloudy, with a high near 88. Chance of precipitation is 60%."
19:40:58.544 ->             },
19:40:58.544 ->             {
19:40:58.578 ->                 "number": 11,
19:40:58.612 ->                 "name": "Thursday Night",
19:40:58.646 ->                 "startTime": "2020-07-30T18:00:00-05:00",
19:40:58.714 ->                 "endTime": "2020-07-31T06:00:00-05:00",
19:40:58.782 ->                 "isDaytime": false,
19:40:58.816 ->                 "temperature": 71,
19:40:58.851 ->                 "temperatureUnit": "F",
19:40:58.885 ->                 "temperatureTrend": null,
19:40:58.919 ->                 "windSpeed": "0 to 5 mph",
19:40:58.987 ->                 "windDirection": "S",
19:40:59.021 ->                 "icon": "https://api.weather.gov/icons/land/night/tsra,60/tsra,40?size=medium",
19:40:59.123 ->                 "shortForecast": "Showers And Thunderstorms Likely",
19:40:59.191 ->                 "detailedForecast": "Showers and thunderstorms likely. Mostly cloudy, with a low around 71. Chance of precipitation is 60%."
19:40:59.328 ->             },
19:40:59.362 ->             {
19:40:59.362 ->                 "number": 12,
19:40:59.396 ->                 "name": "Friday",
19:40:59.430 ->                 "startTime": "2020-07-31T06:00:00-05:00",
19:40:59.498 ->                 "endTime": "2020-07-31T18:00:00-05:00",
19:40:59.566 ->                 "isDaytime": true,
19:40:59.600 ->                 "temperature": 89,
19:40:59.634 ->                 "temperatureUnit": "F",
19:40:59.668 ->                 "temperatureTrend": null,
19:40:59.736 ->                 "windSpeed": "0 mph",
19:40:59.770 ->                 "windDirection": "S",
19:40:59.804 ->                 "icon": "https://api.weather.gov/icons/land/day/tsra_sct,40?size=medium",
19:40:59.906 ->                 "shortForecast": "Chance Showers And Thunderstorms",
19:40:59.974 ->                 "detailedForecast": "A chance of showers and thunderstorms. Mostly cloudy, with a high near 89. Chance of precipitation is 40%."
19:41:00.112 ->             },
19:41:00.146 ->             {
19:41:00.146 ->                 "number": 13,
19:41:00.180 ->                 "name": "Friday Night",
19:41:00.248 ->                 "startTime": "2020-07-31T18:00:00-05:00",
19:41:00.282 ->                 "endTime": "2020-08-01T06:00:00-05:00",
19:41:00.351 ->                 "isDaytime": false,
19:41:00.385 ->                 "temperature": 71,
19:41:00.420 ->                 "temperatureUnit": "F",
19:41:00.454 ->                 "temperatureTrend": null,
19:41:00.522 ->                 "windSpeed": "0 to 5 mph",
19:41:00.556 ->                 "windDirection": "WNW",
19:41:00.590 ->                 "icon": "https://api.weather.gov/icons/land/night/tsra_hi,40/tsra_hi,30?size=medium",
19:41:00.726 ->                 "shortForecast": "Chance Showers And Thunderstorms",
19:41:00.794 ->                 "detailedForecast": "A chance of showers and thunderstorms. Mostly cloudy, with a low around 71. Chance of precipitation is 40%."
19:41:00.930 ->             },
19:41:00.964 ->             {
19:41:00.964 ->                 "number": 14,
19:41:00.998 ->                 "name": "Saturday",
19:41:01.032 ->                 "startTime": "2020-08-01T06:00:00-05:00",
19:41:01.100 ->                 "endTime": "2020-08-01T18:00:00-05:00",
19:41:01.168 ->                 "isDaytime": true,
19:41:01.202 ->                 "temperature": 90,
19:41:01.236 ->                 "temperatureUnit": "F",
19:41:01.270 ->                 "temperatureTrend": null,
19:41:01.338 ->                 "windSpeed": "5 mph",
19:41:01.372 ->                 "windDirection": "WNW",
19:41:01.406 ->                 "icon": "https://api.weather.gov/icons/land/day/tsra_hi,40?size=medium",
19:41:01.508 ->                 "shortForecast": "Chance Showers And Thunderstorms",
19:41:01.576 ->                 "detailedForecast": "A chance of showers and thunderstorms. Mostly sunny, with a high near 90. Chance of precipitation is 40%."
19:41:01.712 ->             }
19:41:01.746 ->         ]
19:41:01.746 ->     }
19:41:06.613 -> }
19:41:11.624 -> 
19:41:16.634 -> 
19:41:21.641 -> 
19:41:26.613 -> 
19:41:31.627 -> 
19:41:36.627 -> 
19:41:41.630 -> 
19:41:46.634 -> 
19:41:51.635 -> 
19:41:56.638 -> 
19:42:01.639 -> 
19:42:06.641 -> 
19:42:11.608 -> 
19:42:16.610 -> 
19:42:21.613 -> 
19:42:26.620 -> 
19:42:31.623 -> 
19:42:36.626 -> 
19:42:41.632 -> 
19:42:46.630 -> 
19:42:51.634 -> 
19:42:56.636 -> 
19:43:01.641 -> 
19:43:06.614 -> 
19:43:11.621 -> 
19:43:16.622 -> 
19:43:21.624 -> 
19:43:26.631 -> 
19:43:31.634 -> 
19:43:36.640 -> 
19:43:41.612 -> 
19:43:46.619 -> 
19:43:51.627 -> 
19:43:56.631 -> 
19:44:01.633 -> 
19:44:06.637 -> 
19:44:11.641 -> 
19:44:16.611 -> 
19:44:21.616 -> 
19:44:26.619 -> 
19:44:31.642 -> 
19:44:36.617 -> 
19:44:41.627 -> 
19:44:46.631 -> 
19:44:51.641 -> 
19:44:56.614 -> 
19:45:01.622 -> 
19:45:06.627 -> 
19:45:11.639 -> 
19:45:16.611 -> 
19:45:21.622 -> 
19:45:26.625 -> 
19:45:31.631 -> 
19:45:36.609 -> 
19:45:41.621 -> 
19:45:46.620 -> 
19:45:51.629 -> 
19:45:56.635 -> 
19:46:01.614 -> 
19:46:06.622 -> 
19:46:11.631 -> 
19:46:16.639 -> 
19:46:21.610 -> 
19:46:26.620 -> 
19:46:31.631 -> 
19:46:36.637 -> 
19:46:41.611 -> 
19:46:46.618 -> 
19:46:51.625 -> 
19:46:56.629 -> 
19:47:01.633 -> 
19:47:06.609 -> 
19:47:11.616 -> 
19:47:16.619 -> 
19:47:21.623 -> 
19:47:26.635 -> 
19:47:31.614 -> 
19:47:36.619 -> 
19:47:41.627 -> 
19:47:46.632 -> 
19:47:51.637 -> 
19:47:56.641 -> 
19:48:01.613 -> 
19:48:06.630 -> 
19:48:11.637 -> 
19:48:16.639 -> 
19:48:21.615 -> 
19:48:26.621 -> 
19:48:31.629 -> 
19:48:36.637 -> 
19:48:41.609 -> 
19:48:46.612 -> 
19:48:51.615 -> 
19:48:56.619 -> 
19:49:11.627 -> 
19:49:12.749 -> Done with Setup
19:49:59.294 -> Time Sent
19:49:59.328 -> ~6 19:50:00
19:50:59.297 -> Time Sent
19:50:59.331 -> ~6 19:51:00

```
